File: src/togo.Api/Controllers/TaskController.cs
using Microsoft.AspNetCore.Mvc;
using togo.Service.Dto;
using togo.Service.Interface;
using System.Threading.Tasks;
using TaskEntity = togo.Service.Context.Task;

namespace togo.Api.Controllers
{
    [Route("tasks")]
    public class TaskController : ControllerBase
    {
        private readonly ITaskService _taskService;
        public TaskController(ITaskService taskService)
        {
            _taskService = taskService;
        }

        [HttpPost]
        public async Task<ApiResponse<TaskEntity>> CreateTask([FromBody] TaskCreateDto input)
        {
            return await _taskService.Create(input);
        }
    }
}

File: src/features/task/task.controller.ts
import { Controller, Post, Body, UseInterceptors } from '@nestjs/common';
import { TaskService } from './task.service';
import { CreateTaskDto } from './dto/create-task.dto';
import { CurrentUserInterceptor } from '../../interceptor/user.interceptor';

@Controller('tasks')
export class TaskController {
  constructor(private readonly taskService: TaskService) {}

  @UseInterceptors(CurrentUserInterceptor)
  @Post()
  async create(@Body() createTaskDto: CreateTaskDto) {
    return await this.taskService.create(createTaskDto);
  }
}
